src/commands/transaction/get.js

```diff
@@ -43,40 +50,110 @@ const queryNode = async (client, txnState, parameters) =>
 
 export default class GetCommand extends BaseCommand {
 	async run() {
-		const { args: { ids }, flags: { state: txnState } } = this.parse(
-			GetCommand,
-		);
-		const req = ids.map(id => ({
-			query: {
-				limit: 1,
-				id,
-			},
-			placeholder: {
-				id,
-				message: 'Transaction not found.',
-			},
-		}));
+		const {
+			args: { ids },
+			flags: { state: txnState, limit, offset, senderId: senderAddress },
+		} = this.parse(GetCommand);
 
 		const client = getAPIClient(this.userConfig.api);
 
-		if (txnState === 'unsigned') {
-			const results = await queryNode(client.node, txnState, req);
+		if (txnState) {
+			if (ids) {
+				const reqTransactionIDs = ids.map(id => ({
+					query: {
+						limit: 1,
+						id,
+					},
+					placeholder: {
+						id,
+						message: 'Transaction not found.',
+					},
+				}));
+
+				const results = await queryNode(
+					client.node,
+					txnState,
+					reqTransactionIDs,
+				);
+				return this.print(results);
+			}
+
+			if (senderAddress) {
```

Comment:
should show some error if provided senderId does not exist or invalid. currently doesn't seem to throw.